src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py

```diff
@@ -11,13 +11,216 @@
+        self.__dirblk = dirblk
+        self.__avgblk = avgblk
+        self.recfm = recfm
+        self.blksize = blksize
+        self.__lrecl = lrecl
+        self.__storclass = storclass
+        self.__mgntclass = mgntclass
+        self.__dataclass = dataclass
+        self.__dsntype = dsntype
+
+    @property
+    def volser(self) -> Optional[str]:
+        return self.__volser
+
+    @volser.setter
+    def volser(self, volser: Optional[str]):
+        self.__volser = volser
+
+    @property
+    def unit(self) -> Optional[str]:
+        return self.__unit
+
+    @unit.setter
+    def unit(self, unit: Optional[str]):
+        self.__unit = unit
+
+    @property
+    def dsorg(self) -> Optional[str]:
+        return self.__dsorg
+
+    @dsorg.setter
+    def dsorg(self, dsorg: Optional[str]):
+        if dsorg not in ("PO", "PS"):
+            if self.like is not None and dsorg is None:
+                pass
+            else:
+                raise ValueError("")
+        self.__dsorg = dsorg
+
+    @property
+    def alcunit(self) -> Optional[str]:
+        return self.__alcunit
+
+    @alcunit.setter
+    def alcunit(self, alcunit: Optional[str]):
+        if alcunit is None:
+            if self.like is not None:
+                self.alcunit = None
+            else:
+                self.__alcunit = "TRK"
+        elif alcunit not in ("CYL", "TRK"):
+            raise KeyError
+        else:
+            self.__alcunit = alcunit
+
+    @property
+    def primary(self) -> Optional[int]:
+        return self.__primary
+
+    @primary.setter
+    def primary(self, primary: Optional[int]):
+        if primary is not None:
+            if primary > 16777215:
+                raise ValueError
+        self.__primary = primary
+
+    @property
+    def secondary(self) -> Optional[int]:
+        return self.__secondary
+
+    @secondary.setter
+    def secondary(self, secondary: Optional[int]):
+        if self.primary is not None:
+            secondary = secondary if secondary is not None else int(self.primary / 10)
+            if secondary > 16777215:
+                raise ValueError
+            self.__secondary = secondary
+
+    @property
+    def dirblk(self) -> Optional[int]:
+        return self.__dirblk
+
+    @dirblk.setter
+    def dirblk(self, dirblk: Optional[int]):
+        self.__dirblk = dirblk
+
+    @property
+    def avgblk(self) -> Optional[int]:
+        return self.__avgblk
+
+    @avgblk.setter
+    def avgblk(self, avgblk: Optional[int]):
+        self.__avgblk = avgblk
+
+    @property
+    def recfm(self) -> Optional[str]:
+        return self.__recfm
+
+    @recfm.setter
+    def recfm(self, recfm: Optional[str]):
+        if recfm is None:
+            if self.like is not None:
+                self.__recfm = None
+            else:
+                self.__recfm = "F"
+        elif recfm not in ("F", "FB", "V", "VB", "U", "FBA", "FBM", "VBA", "VBM"):
+            raise KeyError
```

Comment:
Would it be helpful to list the supported record formats in the error output? This one could go either way, I'm not opposed to omitting the supported formats from the error as there are quite a few options. But, I figured I should mention it as I had left similar comments above.